File: app/src/main/java/com/example/cbanda/achieve/presentation/list/GoalItemFragment.java
package com.example.cbanda.achieve.presentation.list;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.drawable.GradientDrawable;
import android.graphics.drawable.LayerDrawable;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.cbanda.achieve.R;
import com.example.cbanda.achieve.models.db.Goal;
import com.example.cbanda.achieve.presentation.detail.GoalDetailActivity;

import java.util.ArrayList;
import java.util.List;

import static android.content.ContentValues.TAG;


public class GoalItemFragment extends Fragment {


    private static final String ARG_COLUMN_COUNT = "column-count";

    private int mColumnCount = 1;
    GoalListViewModel goalListViewModel;
    GoalsAdapter goalsAdapter;
    RecyclerView recyclerViewGoals;
    LinearLayoutManager mLayoutManager;
    Goal goal;
    private List<Goal> goalList;

    public GoalItemFragment() {
    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);


        goalsAdapter = new GoalsAdapter(new ArrayList<Goal>(), SelectItemClicked());
        setupViewModels();
        //  changePriorityColor();

    }

    public customItemClickListener SelectItemClicked() {
        return new customItemClickListener() {
            @Override
            public void onItemClick(View v, int position) {
                Log.d(TAG, "clicked position:" + position);
                Goal goal = (Goal) v.getTag();
                int id = goal.getId();
                String name = goal.getDescription();
                Intent i = new Intent(getActivity(), GoalDetailActivity.class);
                i.putExtra("Goal", id);
                i.putExtra("Description", name);

                startActivity(i);
            }
        };
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_list_goal, container, false);
        setupRecyclerView(view);
        return view;
    }

    private void setupRecyclerView(View view) {
        recyclerViewGoals = view.findViewById(R.id.recycler_view_goals);
        mLayoutManager = new LinearLayoutManager(this.getActivity().getBaseContext());
        DividerItemDecoration decoration = new DividerItemDecoration(recyclerViewGoals.getContext(), mLayoutManager.getOrientation());
        recyclerViewGoals.addItemDecoration(decoration);
        recyclerViewGoals.setLayoutManager(mLayoutManager);
        recyclerViewGoals.setAdapter(goalsAdapter);


    }

    private void setupViewModels() {

        goalListViewModel = ViewModelProviders.of(getActivity()).get(GoalListViewModel.class);
        LiveData<List<Goal>> goalsLiveDataObject = goalListViewModel.getGoals();

        goalsLiveDataObject.observe(this, new Observer<List<Goal>>() {
            @Override
            public void onChanged(@Nullable List<Goal> goals) {

                goalsAdapter.setGoals(goals);

            }
        });
    }


    public void changePriorityColor() {
        LayerDrawable layerDrawable = (LayerDrawable) getResources()
                .getDrawable(R.drawable.rectangle_stroke);
        GradientDrawable gradientDrawable = (GradientDrawable) layerDrawable
                .findDrawableByLayerId(R.id.gradientDrawable);
        if (goal.getPriority().contains("medium")) {
            gradientDrawable.setColor(Color.GREEN);
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

    }

    @Override
    public void onDetach() {
        super.onDetach();

    }


}

File: app/src/main/java/com/example/cbanda/achieve/presentation/list/GoalItemFragment.kt
package com.example.cbanda.achieve.presentation.list

import android.content.ContentValues.TAG
import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import android.graphics.drawable.LayerDrawable
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.cbanda.achieve.R
import com.example.cbanda.achieve.models.db.Goal
import com.example.cbanda.achieve.presentation.detail.GoalDetailActivity
import com.example.cbanda.achieve.utilities.ViewModelFactory
import java.util.*


class GoalItemFragment : Fragment() {

    private lateinit var goalListViewModel: GoalListViewModel
    private lateinit var viewModelFactory: ViewModelFactory
    private lateinit var goalsAdapter: GoalsAdapter
    private lateinit var recyclerViewGoals: RecyclerView
    private lateinit var mLayoutManager: LinearLayoutManager

    private fun selectItemClicked(): CustomItemClickListener {
        return object : CustomItemClickListener {
            override fun onItemClick(v: View, position: Int) {
                Log.d(TAG, "clicked position:$position")
                val goal = v.tag as Goal
                val id = goal.id
                val name = goal.description
                val intent = Intent(activity, GoalDetailActivity::class.java)
                intent.putExtra("Goal", id)
                intent.putExtra("Description", name)

                startActivity(intent)
            }
        }
    }


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_list_goal, container, false)
        goalsAdapter = GoalsAdapter(ArrayList(), selectItemClicked(), requireContext())
        setupRecyclerView(view)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        setupViewModels()
        getGoals()
    }

    private fun setupRecyclerView(view: View) {
        recyclerViewGoals = view.findViewById(R.id.recycler_view_goals)
        mLayoutManager = LinearLayoutManager(this.activity!!.baseContext)
        val decoration = DividerItemDecoration(recyclerViewGoals.context, mLayoutManager.orientation)
        recyclerViewGoals.addItemDecoration(decoration)
        recyclerViewGoals.layoutManager = mLayoutManager
        recyclerViewGoals.adapter = goalsAdapter


    }

    private fun setupViewModels() {
        goalListViewModel = ViewModelProviders.of(requireActivity()).get(GoalListViewModel::class.java)
    }
     private fun getGoals(){
        val goalsLiveDataObject = goalListViewModel.getGoals()
        goalsLiveDataObject.observe(this, Observer { goals -> goalsAdapter.setGoals(goals) })
    }

}
